src/outcome_spaces/sequential_pair_distances.jl

```diff
@@ -0,0 +1,119 @@
+export SequentialPairDistances
+
+"""
+    SequentialPairDistances <: CountBasedOutcomeSpace
+    SequentialPairDistances(x; n = 3, m = 3, τ = 1, metric = Distances.Chebyshev())
+
+The `SequentialPairDistances` outcome space, which is based on the distribution of 
+distances of sequential pairs of points.
+
+This outcome space appears implicitly as part of the "distribution entropy" introduced
+by [Li2015](@citet), which of course can be reproduced here (see example below).
+We've generalized the method to be used with any [`InformationMeasure`](@ref) and
+[`DiscreteInfoEstimator`](@ref), and with valid distance `metric` (from Distances.jl).
+
+Input data `x` are needed for initialization, because distances must be pre-computed to
+know the minimum/maximum distances needed for binning the distribution of pairwise
+distances. 
+
+## Description
+
+`SequentialPairDistances` does the following: 
+
+- Transforms the input timeseries `x` by first embedding it using embedding dimension
+    `m` and embedding lag `τ`.
```

Comment:
In this case, `x` doesn't need to be given at all to the `SequentialPairDistances`, right?

Reply:
In general I think the design here shold be just like the permutation entropy, including all `codify` business. They do something very similar in the end. Embed -> cast to integer via some encoding.

Reply:
The constructor still needs `x`, because you can't encode without knowing the minimum/maximum distances. So distances between each pair of points must be computed *prior* to encoding, because the fixed rectangular encoding isn't well defined without the min/max distance.

Reply:
Well, strictly speaking, you don't *need* to give it to the constructor. However, if the distances and the encoding is not part of the struct, then they can't be accessed afterwards.

There's pros and cons to this:
- If these fields are not part of the `struct`, then the struct is trivial and it can be used as keys in a dictionary storing results with minimal memory footprint.
- If these fields *are* part of the `struct`, then each instance of the struct can be enormous in terms of memory, and serializing the struct and using it for indexing is hopeless. 

I'm happy with keeping it as is for now. However, I've encountered some issues due to the structs being too large upstream in CausalityTools.jl before, so perhaps we should be a bit wary of taking this approach. Perhaps we can just hide the `encoding` too from the public API for the moment, so we can freely make changes later if necessary?